microprofile-health/src/test/java/org/jboss/eap/qe/microprofile/health/integration/FailSafeCDIHealthBaseTest.java

```diff
@@ -0,0 +1,359 @@
+     * @tpSince EAP 7.4.0.CD19
+     */
+    @Test
+    @RunAsClient
+    public void testLivenessEndpointDownInMaintenace() throws Exception {
+        setConfigProperties(false, true, true, false);
+        get(HealthUrlProvider.liveEndpoint()).then()
+                .statusCode(503)
+                .contentType(ContentType.JSON)
+                .header("Content-Type", containsString("application/json"))
+                .body("status", is("DOWN"),
+                        "checks", hasSize(1),
+                        "checks.status", hasItems("DOWN"),
+                        "checks.name", containsInAnyOrder("dummyLiveness"));
+    }
+
+    /**
+     * @tpTestDetails Multiple-component customer scenario - health check status is based on fail-safe CDI bean and MP Config
+     *                property. There is IOException since service is in maintenance but fallback method should be used.
+     * @tpPassCrit Overall and the health check status is down and fallback method is invoked. MP Metrics are increased
+     *             * according to the specification.
+     * @tpSince EAP 7.4.0.CD19
+     */
+    @Ignore("WFLY-12924, WFLY-12925")
+    @Test
+    @RunAsClient
+    public void testReadinessEndpointDownInMaintenace() throws Exception {
+        setConfigProperties(false, true, true, false);
+        get(HealthUrlProvider.readyEndpoint()).then()
+                .statusCode(503)
+                .contentType(ContentType.JSON)
+                .header("Content-Type", containsString("application/json"))
+                .body("status", is("DOWN"),
+                        "checks", hasSize(1),
+                        "checks.status", hasItems("DOWN"),
+                        "checks.name", containsInAnyOrder("dummyReadiness"));
+
+        MetricsChecker.get()
+                .validateSimulationCounter(FailSafeDummyService.MAX_RETRIES + 1) // 1 call + N retries
+                .validateInvocationsTotal(1)
+                .validateInvocationsFailedTotal(1)
+                .validateRetryRetriesTotal(FailSafeDummyService.MAX_RETRIES) // N retries
+                .validateRetryCallsFailedTotal(1)
+                .validateRetryCallsSucceededTotal(0)
+                .validateFallbackCallsTotal(1);
+
+        get(HealthUrlProvider.readyEndpoint()).then().statusCode(503);
+
+        MetricsChecker.get()
+                .validateSimulationCounter(FailSafeDummyService.MAX_RETRIES * 2 + 2) // 2 calls + 2N retries
+                .validateInvocationsTotal(2)
+                .validateInvocationsFailedTotal(2)
+                .validateRetryRetriesTotal(FailSafeDummyService.MAX_RETRIES * 2) // 2N retries
+                .validateRetryCallsFailedTotal(2)
+                .validateRetryCallsSucceededTotal(0)
+                .validateFallbackCallsTotal(2);
+    }
+
+    /**
+     * Checker for expected MP Metrics values.
+     */
+    public static class MetricsChecker {
```

Comment:
I would welcome a checker which is supplied with response so its scope is clear (to check response, not to make request). But I am OK with this too.

Reply:
yeah, that is better